src/viking.js

```diff
@@ -1,11 +1,119 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+    this.health = health
+    this.strength = strength
+}
+
+Soldier.prototype.attack = function () {
+    return this.strength
+}
+
+Soldier.prototype.receiveDamage = function (damage) {
+    this.health -= damage
+}
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+
+Viking.prototype.receiveDamage = function (damage) {
+    this.health -= damage
+    if (this.health > 0) {
+        return `${this.name} has received ${damage} points of damage`
+    } else {
+        return `${this.name} has died in act of combat`
+    }
+}
+
+Viking.prototype.battleCry = () => {
+    return 'Odin Owns You All!'
+}
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength)
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+
+Saxon.prototype.receiveDamage = function (damage) {
+    this.health -= damage
+    if (this.health > 0) {
+        return `A Saxon has received ${damage} points of damage`
+    } else {
+        return 'A Saxon has died in combat'
+    }
+}
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = []
+    this.saxonArmy = []
+}
+
+War.prototype.addViking = function (viking) {
+    this.vikingArmy.push(viking)
+}
+
+War.prototype.addSaxon = function (saxon) {
+    this.saxonArmy.push(saxon)
+}
+
+War.prototype.vikingAttack = function () {
+    saxonSelected = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
```

Comment:
The code inside this method and the one below are very similar, maybe you could encapsulated in a function and reused.